File: golang/testsuite/testsuite_impl/example_testsuite.go
package testsuite_impl

import (
	"github.com/kurtosis-tech/kurtosis-libs/golang/testsuite/testsuite_impl/advanced_network_test"
	"github.com/kurtosis-tech/kurtosis-libs/golang/testsuite/testsuite_impl/basic_datastore_and_api_test"
	"github.com/kurtosis-tech/kurtosis-libs/golang/testsuite/testsuite_impl/basic_datastore_test"
	"github.com/kurtosis-tech/kurtosis-testsuite-api-lib/golang/lib/testsuite"
)

type ExampleTestsuite struct {
	apiServiceImage string
	datastoreServiceImage string
}

func NewExampleTestsuite(apiServiceImage string, datastoreServiceImage string) *ExampleTestsuite {
	return &ExampleTestsuite{apiServiceImage: apiServiceImage, datastoreServiceImage: datastoreServiceImage}
}

func (suite ExampleTestsuite) GetTests() map[string]testsuite.Test {
	tests := map[string]testsuite.Test{
		"basicDatastoreTest": basic_datastore_test.NewBasicDatastoreTest(suite.datastoreServiceImage),
		"basicDatastoreAndApiTest": basic_datastore_and_api_test.NewBasicDatastoreAndApiTest(
			suite.datastoreServiceImage,
			suite.apiServiceImage,
		),
		"advancedNetworkTest": advanced_network_test.NewAdvancedNetworkTest(
			suite.datastoreServiceImage,
			suite.apiServiceImage,
		),
	}
	return tests
}

File: typescript/testsuite/testsuite_impl/example_testsuite.ts
import { AdvancedNetworkTest } from "./advanced_network_test/advanced_network_test";
import { BasicDatastoreAndApiTest } from "./basic_datastore_and_api_test/basic_datastore_and_api_test";
import { BasicDatastoreTest } from "./basic_datastore_test/basic_datastore_test";
import { Test } from "kurtosis-testsuite-api-lib";


export class ExampleTestsuite {
    private readonly apiServiceImage: string;
    private readonly datastoreServiceImage: string;

    constructor(apiServiceImage: string, datastoreServiceImage: string) {
        this.apiServiceImage = apiServiceImage;
        this.datastoreServiceImage = datastoreServiceImage;
    }

    public getTests(): Map<string, Test> {
        const tests: Map<string, Test> = new Map();
        tests.set("basicDatastoreTest", new BasicDatastoreTest(this.datastoreServiceImage));
        tests.set("basicDatastoreAndApiTest", new BasicDatastoreAndApiTest(this.datastoreServiceImage, this.apiServiceImage));
        tests.set("advancedNetworkTest", new AdvancedNetworkTest(this.datastoreServiceImage, this.apiServiceImage));
        return tests;
    }
}
